### ISSUE TITLE
Binary Cross Entropy loss is incorrect: author confused about what BCE is

### ISSUE THREAD

**Issue by mcourteaux** [keras-team/tf-keras#223]
**Describe the problem**.

The Binary Cross Entropy loss implementation computes the mean (along an axis of choice) of the binary cross entropy loss. Obviously, this mean is pointless, and has caused a lot of confusion with the people writing the documentation it seems.

The mean reduction I'm talking about is at the last line of this function (L2155):
https://github.com/keras-team/keras/blob/f9336cc5114b4a9429a242deb264b707379646b7/keras/losses.py#L2116-L2158

So clearly, judging by its name, this function promises to calculate the _binary crossentropy_. However, the reduction axis there is very weird, and the author clearly got confused with the more general crossentropy definition (where you indeed take the expected value of `-log(x)`. In other words, with a summation, you take the sum over `-x*log(x)`, with `x` being the probabilities summing to 1), like here (from [Wiki](https://en.wikipedia.org/wiki/Cross_entropy#Definition)):
![image](https://user-images.githubusercontent.com/845012/229758225-30d1ed6c-9905-4f72-be07-f17095570786.png)

However, the mistake is that the `backend.binary_crossentropy` function already performs this reduction internally:

https://github.com/keras-team/keras/blob/f9336cc5114b4a9429a242deb264b707379646b7/keras/backend.py#L5687-L5717

The summation is in the fact that the implementation computes `bce` with the two terms (one in `bce = ...` at L5715, and one in `bce += ...` at L5716). These two terms are the summation:
![image](https://user-images.githubusercontent.com/845012/229758225-30d1ed6c-9905-4f72-be07-f17095570786.png)

So, as an intermediate conclusion: `keras.backend.binary_crossentropy` has the correct implementation and does what it promises.

Now, the documentation of `keras.metrics.binary_crossentropy()` is suggesting to use it in a pointless way:

https://github.com/keras-team/keras/blob/f9336cc5114b4a9429a242deb264b707379646b7/keras/losses.py#L2121-L2126

So, clearly, the writer of the function and documentation thought there was a need to supply the full list of probabilities that sum to 1. All pairs in this demonstration sum to one (0.6 + 0.4, and 0 + 1). Then, the `backend.binary_crossentropy()` is called on these inputs, which computes 4 (four!) binary crossentropies, which has a resulting shape of `(2, 2)` (just like the input shape!), which is then mean-reduced over the last axis to finally yield `(2,)` as a shape.

Now clearly, the above described example from the documentation is just a demonstration of how to use multi-class crossentropy, applied on a case with only 2 classes. Under the hood, the function calculates the binary crossentropy twice for every input, and is then averaged out. The final output of the example is correct, but the whole function is completely bananas.

**Describe the current behavior**.

The function does not correctly compute binary cross-entropy if you do not provide it with pairs of probabilities that sum to 1.

**Describe the expected behavior**.

It just computes the actual binary cross-entropy, and the mean-reduction is thrown out.

A correct documentation example would be:

    Standalone usage:
    >>> y_true = [1, 0]
    >>> y_pred = [0.6, 0.4]
    >>> loss = tf.keras.losses.binary_crossentropy(y_true, y_pred) # No axis nonsense
    >>> assert loss.shape == (2,)

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing): remove the backend.mean call, remove the axis parameter. This probably severely breaks whatever it is that people have built around this mistake (either depend on it, or a workaround), but long-term, it's the right thing to do.
